### PR TITLE
Devise

### PR DESCRIPTION
**Pull request by Thibault-Sordet**

### CODE REVIEW

File: app/models/booking.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class Booking < ApplicationRecord
+  belongs_to :star
+  belongs_to :user
+  belongs_to :review
```
Comment (NicolasNeiman) [inline app/models/booking.rb]: has_one :review or has_multiple :reviews

File: app/models/review.rb [hunk 2]
```diff
@@ -0,0 +1,3 @@
+class Review < ApplicationRecord
+  validates :rating, presence: true
```
Comment (Joannacha) [inline app/models/review.rb:3]: `belongs_to :booking`

Comment (NicolasNeiman) [inline app/models/review.rb:3]: belongs_to :booking

File: app/models/star.rb [hunk 3]
```diff
@@ -0,0 +1,6 @@
+class Star < ApplicationRecord
+  belongs_to :user
+  has_many :bookings
+  validates :name, presence: true
```
Comment (Joannacha) [inline app/models/star.rb]: add `uniqueness: true`

File: app/models/star.rb [hunk 4]
```diff
@@ -0,0 +1,6 @@
...
+  validates :price, presence: true
```
Comment (Joannacha) [inline app/models/star.rb]: `numericality: { only_integer: true }`

File: app/models/user.rb [hunk 5]
```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatabl
```
Comment (NicolasNeiman) [inline app/models/user.rb]: validatablE

File: app/models/star.rb [hunk 6]
```diff
@@ -0,0 +1,6 @@
+class Star < ApplicationRecord
+  belongs_to :user
+  has_many :bookings
```
Comment (Joannacha) [inline app/models/star.rb:3]: `has_many :reviews, :through :bookings`

File: db/schema.rb [hunk 7]
```diff
@@ -10,9 +10,55 @@
...
+    t.datetime "created_at", precision: 6, null: false
+    t.datetime "updated_at", precision: 6, null: false
+    t.index ["review_id"], name: "index_bookings_on_review_id"
+    t.index ["star_id"], name: "index_bookings_on_star_id"
+    t.index ["user_id"], name: "index_bookings_on_user_id"
+  end
+
+  create_table "reviews", force: :cascade do |t|
+    t.integer "rating"
+    t.text "comment"
```
Comment (NicolasNeiman) [inline db/schema.rb:32]: missing booking ?

**Review by Joannacha — CHANGES_REQUESTED** [review]
left 4 comments to be reviewed

File: app/models/review.rb (hunk 2 above)
Comment (Thibault-Sordet) [reply app/models/review.rb:3]: done

File: app/models/star.rb (hunk 3 above)
Comment (Thibault-Sordet) [reply app/models/star.rb]: done

File: app/models/star.rb (hunk 4 above)
Comment (Thibault-Sordet) [reply app/models/star.rb]: done

File: app/models/review.rb (hunk 2 above)
Comment (Thibault-Sordet) [reply app/models/review.rb:3]: done

File: app/models/user.rb (hunk 5 above)
Comment (Thibault-Sordet) [reply app/models/user.rb]: done

File: db/schema.rb (hunk 7 above)
Comment (Thibault-Sordet) [reply db/schema.rb:32]: done

File: app/models/star.rb (hunk 6 above)
Comment (Thibault-Sordet) [reply app/models/star.rb:3]: done

**Comment by Thibault-Sordet** [comment]
Implement the devise signin/signup for user plus the model star, review, booking

**Pull request merged**